File: backend/java-spring/src/main/java/com/unosquare/worldcup/model/ItineraryStop.java
package com.unosquare.worldcup.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.UUID;

@Entity
@Table(name = "itinerary_stops")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class ItineraryStop {

    @Id
    private String id;
    private int stopNumber;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "match_id")
    private Match match;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "city_id")
    private City city;

    private double distanceFromPrevious;

    @ManyToOne
    @JoinColumn(name = "itinerary_id")
    @JsonIgnore
    private Itinerary itinerary;

    @PrePersist
    public void prePersist() {
        if (id == null) id = UUID.randomUUID().toString();
    }
}

File: backend/dotnet-webapi/src/WorldCup.Api/Models/ItineraryStop.cs
namespace WorldCup.Api.Models;

public class ItineraryStop
{
    public string Id { get; set; } = Guid.NewGuid().ToString();
    public string ItineraryId { get; set; } = string.Empty;
    public int StopNumber { get; set; }
    public string MatchId { get; set; } = string.Empty;
    public string CityId { get; set; } = string.Empty;
    public double DistanceFromPrevious { get; set; }

    // Navigation properties
    public Itinerary Itinerary { get; set; } = null!;
    public Match Match { get; set; } = null!;
    public City City { get; set; } = null!;
}
